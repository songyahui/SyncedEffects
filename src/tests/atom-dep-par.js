"use hopscript"

const hh = require("hiphop")

const prg =
      <hh.module A=${{combine: (x, y) => x + y}}>
	<hh.parallel>
	  <hh.loop>
	    <hh.emit A value=${0} />
	    <hh.pause/>
	  </hh.loop>

	  <hh.loop>
	    <hh.emit A value=${1} />
	    <hh.atom apply=${function() {
	       console.log(this.A.nowval);
	    }}/>
	    <hh.pause/>
	  </hh.loop>

	  <hh.loop>
	    <hh.emit A value=${2} />
	    <hh.atom apply=${function() {
	       console.log(this.A.nowval);
	    }}/>
	    <hh.pause/>
	  </hh.loop>
	</hh.parallel>
    </hh.module>;

let machine = new hh.ReactiveMachine(prg, "error2");

machine.debug_emitted_func = console.log;
machine.react()
machine.react()
machine.react()
machine.react()
machine.react()
