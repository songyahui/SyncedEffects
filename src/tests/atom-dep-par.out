3
3
[ 'A(3)' ]
3
3
[ 'A(3)' ]
3
3
[ 'A(3)' ]
3
3
[ 'A(3)' ]
3
3
[ 'A(3)' ]
